packages/server/lambdas/alfresco/upload-file.ts

```diff
@@ -64,7 +72,18 @@ export async function handleRequest(event: ALBEvent): Promise<ALBResult | undefi
     const headers = (await getAlfrescoOptions(user.uid)).headers;
     const requestOptions = (await fileRequestBuilder(event, headers)) as RequestInit;
 
-    const result = await postFile(requestOptions, categoryData.alfrescoFolder);
+    let result;
+    if (!nestedFolderId) {
+      result = await postFile(requestOptions, categoryData.alfrescoFolder);
+    } else {
+      const foundFolder = await getFolder(user.uid, nestedFolderId);
+      const folderPath = get(foundFolder, 'entry.path.name', '');
+      // Check if the nest folder is a descendant of the category
+      const isFolderDescendantOfCategory = await isFolderInCategory(folderPath, category);
+      if (isFolderDescendantOfCategory) {
```

Comment:
Should this throw an error, if it's not a descendant? Because in that case, this seems to be throwing an error for failing to JSON.stringify the empty result. Throwing something more descriptive would be more useful for possible debugging.